ui/src/app/applications/components/application-pod-view/pod-view.tsx

```diff
@@ -0,0 +1,403 @@
+import {DataLoader, DropDown, DropDownMenu, MenuItem, NotificationType, Tooltip} from 'argo-ui';
+import * as PropTypes from 'prop-types';
+import * as React from 'react';
+import {Checkbox as ReactCheckbox} from 'react-form';
+import Moment from 'react-moment';
+import {EmptyState, ErrorNotification} from '../../../shared/components';
+import {AppContext} from '../../../shared/context';
+import {Application, ApplicationTree, InfoItem, InfraNode, Metric, Pod, ResourceList, ResourceName, ResourceNode, ResourceStatus} from '../../../shared/models';
+import {PodViewPreferences, services, ViewPreferences} from '../../../shared/services';
+import {ResourceTreeNode} from '../application-resource-tree/application-resource-tree';
+import {ResourceIcon} from '../resource-icon';
+import {ResourceLabel} from '../resource-label';
+import {nodeKey, PodHealthIcon} from '../utils';
+import {ComparisonStatusIcon, HealthStatusIcon} from '../utils';
+import './pod-view.scss';
+
+interface PodViewProps {
+    tree: ApplicationTree;
+    onItemClick: (fullName: string) => void;
+    app: Application;
+    nodeMenu?: (node: ResourceNode) => React.ReactNode;
+}
+
+export type PodGroupType = 'topLevelResource' | 'parentResource' | 'node';
+
+export interface PodGroup extends Partial<ResourceNode> {
+    type: PodGroupType;
+    pods: Pod[];
+    info?: InfoItem[];
+    metrics?: ResourceList;
+    resourceStatus?: Partial<ResourceStatus>;
+    renderMenu?: () => React.ReactNode;
+    fullName?: string;
+}
+
+export class PodView extends React.Component<PodViewProps> {
+    private get appContext(): AppContext {
+        return this.context as AppContext;
+    }
+    public static contextTypes = {
+        apis: PropTypes.object
+    };
+    private loader: DataLoader;
+    public render() {
```

Comment:
Please fix React warning  `Each child in a list should have a unique "key" prop` in this component.